### PR TITLE
Cover block: paid-block-media-placeholder fix zero height

### LINKED ISSUE #22152: Cover block: paid-block-media-placeholder sets zero height to block placeholder
### Impacted plugin

Jetpack

### Steps to Reproduce

1. Create a new post on a free WP.com site
2. Add Cover block
3. Add any other block(s)
4. See how they're hidden behind the Cover block placeholder as the `paid-block-media-placeholder` sets the absolute position with 0 heigh.

### A clear and concise description of what you expected to happen.

Other blocks should not be hiding behind the Cover block placeholder.

### What actually happened

_No response_

### Other information

The `paid-block-media-placeholder` was an issue before (#19488), where it was standing in the way of the actual Cover block content. It was fixed in #19489 via the `.paid-block-media-placeholder:not(:only-child)` selector which was correctly intended to not be applied in a placeholder state. At some point (some Gutenberg upgrade) though, a `.components-resizable-box__container` element has been added below the `.paid-block-media-placeholder`, which made the `.paid-block-media-placeholder:not(:only-child)` selector invalid and broke the block's actual placeholder as a result. 

Updating the [current selector](https://github.com/Automattic/jetpack/blob/56dc76f857357bec0bde3c98e8f640eb5d32fd4c/projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/editor.scss#L118-L131) to the following one seems to be fixing the issue:

```css
.wp-block-cover:not(.is-placeholder) .paid-block-media-placeholder { ... }
```

https://user-images.githubusercontent.com/1451471/146987643-362daf33-4a73-4bd8-ac0f-575da3edfabe.mov

cc @ciampo

### Operating System

macOS

### OS Version

_No response_

### Browser

Chrome/Chromium

### Browser Version(s)

_No response_

### PR DESCRIPTION
**Pull request by amustaque97**
Fixes #22152 

#### Changes proposed in this Pull Request:

* Fixed the css to avoid overlapping

#### Testing instructions:

* checkout this branch and build project
* Create a new post on a free WP.com site
* Add Cover block
* Add any other block(s)
* See they're not hidding behind the Cover block placeholder as the paid-block-media-placeholder 

cc @ciampo @WunderBart

### CODE REVIEW

**Comment by ciampo** [comment]
Thank you @amustaque97 ! I won't be able to review this before the end of this week. Tagging also @WunderBart in case he has any capacity to review before me — but no big deal in case you don't!

**Comment by jeherve** [comment]
Internal reference: D72985-code

File: projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/editor.scss [hunk 1]
```diff
@@ -121,7 +121,7 @@ $icon-background-color: #fff;
 
 	// Remove paid-block-media-placeholder from being a flexbox child
 	// when the cover block has a background set
-	.wp-block-cover &:not( :only-child ) {
+	.wp-block-cover :not( .is-placeholder ) {
```
Comment (ciampo) [inline projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/editor.scss:124]: I see a couple of issues with the way this CSS selector is written:

- the `:not( .is-placeholder )` selector should be applied to `.wp-block-cover` (i.e. no space before the `:not`)
- we should add a `&` at the end of the selector, to make sure that the styles apply to `.paid-block-media-placeholder`


```suggestion
	.wp-block-cover:not( .is-placeholder ) & {
```

File: projects/plugins/jetpack/changelog/fix-22152-paid-block-media-placeholder [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: bugfix
+
+Other blocks should not be hiding behind the Cover block placeholder.
```
Comment (ciampo) [inline projects/plugins/jetpack/changelog/fix-22152-paid-block-media-placeholder:4]: Let's reword this to be a bit more informative, maybe 

```suggestion
Cover block: avoid the block's placeholder rendering on top of other blocks
```

**Review by ciampo — COMMENTED** [review]
Hey @amustaque97, feel free to ignore the message above.

In the meantime, I went ahead and tested your changes. The original issue is fixed, but a new regression is also introduced — when adding text to the cover block, the text does align correctly:

Screenshot 2022-01-19 at 20 02 08

I left a suggestion on how to potentially remove the regression

**Comment by ciampo** [comment]
Changes in D73393-code have been applied to wpcom

**Review by jeherve — APPROVED** [review]
